### ISSUE TITLE
svelte-canvas

### ISSUE THREAD

**Issue by dnass** [aragon999/idle-td#1]
Hey! I'm the author of `svelte-canvas`. Apologies for snooping, I just wanted to say it's great to see you're using the library, and if you run into any bugs, pain points, or ideas, just let me know.

**Comment by aragon999** [comment]
Hey there, I am happy that you found the repository :-)

I am really happy for the library, it has been a pleasure to use so far. Most problems I had were related to not properly understanding the concepts of Svelte (this is my first project with Svelte).

In the beginning I had an issue with image drawing on a canvas. If the `Image` resource was not yet loaded. A redraw, after the image was fully loaded, was needed. 
Maybe it would be nice to automate this, but the only solution I can think of is to create a predefined, e.g. `<CanvasImage />`-element.
In general this might be also a good idea, to have for several shapes and elements, but I am not sure if this is the aim of the library or should be in a separate library building upon `svelte-canvas`.

But maybe I also have some misunderstanding and also using `svelte-canvas` in the wrong way, for example the [`TowerLayer`](https://github.com/aragon999/idle-td/blob/master/src/Tower/TowerLayer.svelte) renders each `Tower` as individual [`Layer`](https://github.com/aragon999/idle-td/blob/master/src/Tower/Tower.svelte#L6). Or if it should be more like that I have one `TowerLayer` where the renderer of the `Layer` takes care to render all towers. I saw that in the examples the scatter plot works in a similar way. 
The solution I currently choose is more flexible in terms of developer experience, but maybe it is not how you work with a canvas.
This is definitely something I will investigate in the future.

**Comment by dnass** [comment]
Hey, I totally thought I responded to this a week ago but it seems I didn't. I think your approach for `TowerLayer` is right on. After a certain point, having a ton of individual layers might be less performant than fewer, bigger layers, but there's no point optimizing for that if you're not seeing any issues.

Re images, one idea might be to create a class that manages the asset loading — that way you'd only need one event listener per image, rather than one per tile. The manager class could pass on a placeholder until the image loads, and `FieldNode` could naively render whatever `background` it receives.

I've definitely thought about adding some built-in layer types, but for now I've decided to keep the library simple and unopinionated. I appreciate the feedback though!
